### ISSUE TITLE
Configuration heirarchy not working

### ISSUE THREAD

**Issue by anshulguleria** [eslint/eslint#4986]
I am using `linux 15.04` and `eslint 1.10.3`

I had one `.eslintrc` file in my project's root folder and one in my home directory of linux but the rules in my project's root folder `.eslintrc` file are picked and when i remove file from my projects root folder then rules form home `.eslintrc` file are picked. But I expected that the rules would be merged and rules in my project's root `.eslintrc` file would take preference.

I am using `vim` editor with `syntastic plugin` for handling linting in my editor.

```js
// project's .eslintrc file
{
    "globals": {
        "DRP": false,
        "$": false
    }
}
```

```js
// home .eslintrc file
{
    "parser": "babel-eslint",
    "env": {
        "browser": true,
        "node": true,
        "es6": true
    },
    "ecmaFeatures": {
        "blockBindings": true,
        "forOf": true,
        "templateStrings": true,
        "module": true
    },
    "rules": {
        "strict": 0,
        "eqeqeq": 1,
        "curly": 1,
        "quotes": 0,
        "no-unused-vars": 1,
        "no-dupe-args": 2,
        "no-dupe-keys": 2,
        "no-redeclare": [2, {"builtinGlobals": true}],
        "no-undef": 2
    }
}
```

**Comment by eslintbot** [comment]
@anshulguleria Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
Home directory is a special case. We don't ever read config file from home directory if local configuration files are found. If we can't find any local config files, then we fall back to home directory. This is working as intended. You can read more about config cascading here: http://eslint.org/docs/user-guide/configuring#configuration-cascading-and-hierarchy

Closing as this is working as intended. Feel free to reopen if you feel that this is not working.
